File: Searching-And-Sorting/Bubble_Sort.rb
def bubble_sort(arr)
  loop do
    flag = false

    arr[0...-1].each_index do |i|
      if arr[i] > arr[i+1]
        arr[i], arr[i+1] = arr[i+1], arr[i]
        flag = true
      end
    end

    break if flag == false
  end

  arr
end

# For potentially better readability
def bubble_sort(arr)
  loop do
    sorted = true

    arr[0...-1].each_index do |i|
      if arr[i] > arr[i+1]
        sorted = false
        arr[i], arr[i+1] = arr[i+1], arr[i]
      end
    end

    break if sorted == true
  end

  arr
end

File: Searching-And-Sorting/bubbleSort.js
function bubbleSort(arr) {
  while (true) {
    let sorted = true;

    for (let i = 0; i < arr.length - 1; i++) {
      if (arr[i] > arr[i+1]) {
        sorted = false;
        let temp = arr[i];
        arr[i] = arr[i+1];
        arr[i+1] = temp;
      }
    }

    if (sorted === true) {
      break;
    }
  }

  return arr;
}
